### ISSUE TITLE
Support Gaussian Multivariate Likelihood

### ISSUE THREAD

**Issue by cemoody** [pfnet/chainer#575]
The multivariate Gaussian likelihood is a key ingredient in many models (correlated LDA and mixed-effects models). It's useful in crafting a latent space whose components aren't necessarily independent and are correlated through a latent covariance matrix. 

The gradient isn't hard, but does require a [matrix inversion] which we don't currently support in CuPy.(http://stats.stackexchange.com/questions/27436/how-to-take-derivative-of-multivariate-normal-density)

**Comment by delta2323** [comment]
Thank you for suggestion. My first thought is to implement the variant of `gaussian_nll` which acceptsnon-diagonal matrix as covariant matrices. Do you think it is enough?

**Comment by cemoody** [comment]
Yup! `gaussian_nll` with a non-diagonal covariance structure would be amazing. I think, but I'm not entirely sure, that this feature would allow users to write MAP-estimates to correlated LDA, mixed-effects models, Gaussian Processes and more powerful variational autoencoders that don't have to assume independence in the latent space. I think one of the big difficulties here is that it could depend on other Chainer functions like a determinant and a gradient of a determinant, the inverse of a matrix, and then the gradient of that too -- which we currently don't have. Unfortunately writing the associated kernels is beyond my CUDA & CuPy abilities, but does sound fun. :)

**Comment by beam2d** [comment]
Is it now available, since the two PRs are merged?
